pkgs/tools/archivers/cpio/default.nix

```
@@ -17,6 +17,11 @@ in stdenv.mkDerivation {
 
   enableParallelBuilding = true;
 
+  # GCC 10 changed this default, causing cpio not to compile. This was
+  # necessary as of Jan 2021 (version 2.13); if a new release comes out, check
+  # if it's still needed and remove it if not.
+  NIX_CFLAGS_COMPILE = "-fcommon";
```

The darwin stdenv contains cpio, but we can avoid the rebuild:
```suggestion
  NIX_CFLAGS_COMPILE = if stdenv.cc.isGNU then "-fcommon" else null;
```

---

Also it looks like gentoo has a fix for this: https://bugs.gentoo.org/705900

---

It goes to staging anyway so we don't really need to worry about stdenv rebuilds.

---

Aha, that solves the Darwin mystery. My inclination is to use CFLAGS instead of patches initially, because it keeps the PR simple and seems less likely to break things; we can always switch to patches on a per-package basis later. But I can probably be convinced otherwise.